x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/index/fielddata/TriangleTreeReader.java

```diff
@@ -49,7 +54,7 @@ private static boolean visit(
             }
             thisMinX = x;
             thisMinY = y;
-        } else if ((metadata & 1 << 3) == 1 << 3) { // component in this node is a line
+        } else if ((metadata & LINE) == LINE) { // component in this node is a line
```

Comment:
With your improvements, the comments become less necessary. I think they were there before because without them it was completely unclear what was meant.

Reply:
I think they still not hurt :)